src/AWS.Deploy.Orchestration/DeploymentSettingsHandler.cs

```diff
@@ -91,5 +100,54 @@ public async Task ApplySettings(DeploymentSettings deploymentSettings, Recommend
 
             throw new InvalidDeploymentSettingsException(DeployToolErrorCode.DeploymentConfigurationNeedsAdjusting, errorMessage.Trim());
         }
+
+        public async Task SaveSettings(string filePath, Recommendation recommendation, CloudApplication cloudApplication, OrchestratorSession orchestratorSession)
+        {
+            var parentDirectory = Path.GetDirectoryName(filePath);
+            if (string.IsNullOrEmpty(parentDirectory))
+            {
+                var message = $"Failed to save deployment settings because {filePath} is not a valid path";
+                throw new FailedToSaveDeploymentSettingsException(DeployToolErrorCode.FailedToSaveDeploymentSettings, message);
+            }
+            if (!_directoryManager.Exists(parentDirectory))
```

Comment:
These 2 validations should have happened before the deployment even started. Not saying we can't also validate here is as well.